FlowCal/excel_ui.py

```diff
@@ -773,17 +781,29 @@ def generate_histograms_table(samples_table, samples):
             if pd.notnull(samples_table[header][sample_id]):
                 # Get units in which bins are being reported
                 unit = samples_table[header][sample_id]
-                # Store bins
-                bins = sample.domain(channel)
+                # Decide whether to produce histograms in linear or log scale
+                log = (unit != 'Channel') and \
+                    bool(sample.amplification_type(channel)[0])
+                # Define number of bins
+                nbins = min(sample.resolution(channel), max_bins)
+                # Calculate bin edges
+                bin_edges = sample.hist_bins(channel, nbins, log)
+                # Calculate bin centers
+                if log:
+                    bin_centers = (np.log10(bin_edges[:-1]) + \
+                        np.log10(bin_edges[1:]))/2
+                    bin_centers = 10**bin_centers
+                else:
+                    bin_centers = (bin_edges[:-1] + bin_edges[1:])/2
+                # Store bin centers
                 hist_table.loc[(sample_id,
                                 channel,
-                                'Bin Values ({})'.format(unit)),
-                               columns[0:len(bins)]] = bins
+                                'Bin Centers ({})'.format(unit)),
+                                columns[0:len(bin_centers)]] = bin_centers
                 # Calculate and store histogram counts
-                bin_edges = sample.hist_bin_edges(channel)
                 hist, __ = np.histogram(sample[:,channel], bins=bin_edges)
                 hist_table.loc[(sample_id, channel, 'Counts'),
-                               columns[0:len(bins)]] = hist
+                               columns[0:len(bin_centers)]] = hist
```

Comment:
Previously we used "Bin values", but as you said before this is poorly defined. I changed the description to "bin centers` because I think this generalizes better to when the data has a resolution of 2e5 but we use a much lower number of bins (what is the "bin value" then?).

Yeah this is for plotting. I'm not against reporting bin edges instead of bin centers, but this should be another issue. We would have to deal with the fact that the list of bin edges will contain one more value than the event counts, and how to present this to the user in a way that is not confusing.